scripts/cat/status.py

```
@@ -0,0 +1,639 @@
+            "standing_history": self.standing_history,
+        }
+
+    def generate_new_status(
+        self,
+        age: CatAge = None,
+        social: CatSocial = None,
+        group: CatGroup = None,
+        rank: CatRank = None,
+    ):
+        """
+        Starts a group history and standing history for a newly generated cat. You MUST include either age or rank.
+        :param age: The age the cat currently is.
+        :param social: The social group the cat will be (rogue, clancat, loner, kittypet)
+        :param group: The group the cat will be part of, default is None. If social is set to clancat and group is None,
+         group will default to player clan.
+        :param rank: The rank the cat holds within a group. If they have no group, then this matches their social.
+        """
+        # just some extra checks in case a str snuck in
+        group, rank, social = self.get_enums(group, rank, social, age)
+
+        self._start_group_history(
+            age,
+            social,
+            group,
+            rank,
+        )
+
+        self._start_standing()
+
+    def _start_group_history(
+        self,
+        age: CatAge = None,
+        social: CatSocial = None,
+        group: CatGroup = None,
+        rank: CatRank = None,
+    ):
+        """
+        Generates initial group history for a cat
+        You HAVE to include either an age or a rank for this to work correctly
+        :param age: The age of the cat.
+        :param social: The social standing of the cat (rogue, loner, clancat, ect.)
+        :param group: The group this cat belongs to
+        :param rank: This cat's rank. If the cat is outside the Clan, this will match it's social.
+        """
+        new_history = {"group": group, "rank": rank, "moons_as": 0}
+
+        if not age and not rank:
+            print(
+                "WARNING: group history could not be made due to missing age and rank information"
+            )
+            return
```

I think this should raise an exception rather than just silent return.